### PR TITLE
ESQL: Rename AUTO_BUCKET to just BUCKET

### PR DESCRIPTION
**Pull request by bpintea**
This renames the function `AUTO_BUCKET` to just `BUCKET`.

It also removes the experimental tagging of the function in the docs, making it generally available.

### CODE REVIEW

File: docs/reference/esql/functions/date-time-functions.asciidoc [hunk 1]
```diff
@@ -8,7 +8,7 @@
 {esql} supports these date-time functions:
 
 // tag::date_list[]
-* experimental:[] <<esql-auto_bucket>>
+* <<esql-bucket>>
```
Comment (alex-spies) [inline docs/reference/esql/functions/date-time-functions.asciidoc:11]: Is it intended that we declare `BUCKET` as released in this PR?

(If so, let's add this to the PR description as it's important IMHO.)

**Review by alex-spies — COMMENTED** [review]
Looks good - just one, potentially important remark :)

File: docs/reference/esql/functions/date-time-functions.asciidoc (hunk 1 above)
Comment (bpintea) [reply docs/reference/esql/functions/date-time-functions.asciidoc:11]: > Is it intended that we declare BUCKET as released in this PR?

Yes, I believe we wanted to decide on the naming, also considering the non-auto version.

**Review by costin — APPROVED** [review]
LGTM however please wait for Andrei's review.

**Review by astefan — APPROVED** [review]
LGTM

**Comment by bpintea** [comment]
Thanks, Alex, Costin, Andrei.

**Review by alex-spies — APPROVED** [review]

**Pull request merged**